src/main/java/graphql/execution/directives/QueryDirectives.java

```diff
@@ -66,6 +68,16 @@ public interface QueryDirectives {
      */
     Map<Field, List<QueryAppliedDirective>> getImmediateAppliedDirectivesByField();
 
+    /**
+     * This will return a map of {@link QueryAppliedDirective} to a map of their argument values in {@link NormalizedInputValue} form
+     * <p>
+     * NOTE : This will only be available when {@link graphql.normalized.ExecutableNormalizedOperationFactory} is used
+     * to create the {@link QueryAppliedDirective} information
+     *
+     * @return a map of applied directive to named argument values
+     */
+    Map<QueryAppliedDirective, Map<String, NormalizedInputValue>> getNormalizedInputValueByImmediateAppliedDirectives();
```

Comment:
for a given QueryAppliedDirective - we can get a normalised view of its argument values